packages/sync/src/replicastore/class-table-checksum.php

```diff
@@ -0,0 +1,524 @@
+class Table_Checksum {
+
+	/**
+	 * Table to be checksummed.
+	 *
+	 * @var string
+	 */
+	public $table = '';
+
+	/**
+	 * Table Checksum Configuration.
+	 *
+	 * @var array
+	 */
+	public $table_configuration = array();
+
+	/**
+	 * Field to be used for range queries.
+	 *
+	 * @var string
+	 */
+	public $range_field = '';
+
+	/**
+	 * ID Field(s) to be used.
+	 *
+	 * @var array
+	 */
+	public $key_fields = array();
+
+	/**
+	 * Field(s) to be used in generating the checksum value.
+	 *
+	 * @var array
+	 */
+	public $checksum_fields = array();
+
+	/**
+	 * SQL Query to be used to filter results (allow/disallow).
+	 *
+	 * @var string
+	 */
+	public $additional_filter_sql = '';
+
+	/**
+	 * Default Checksum Table Configurations.
+	 *
+	 * @var array
+	 */
+	public $default_tables = array();
+
+	/**
+	 * Salt to be used when generating checksum.
+	 *
+	 * @var string
+	 */
+	public $salt = '';
+
+	/**
+	 * Tables which are allowed to be checksummed.
+	 *
+	 * @var string
+	 */
+	public $allowed_tables = array();
+
+	/**
+	 * Table_Checksum constructor.
+	 *
+	 * @param string $table
+	 * @param string $salt
+	 *
+	 * @throws Exception
+	 */
+	public function __construct( $table, $salt = null ) {
+		$this->salt = $salt;
+
+		$this->default_tables = $this->get_default_tables();
+
+		// TODO change filters to allow the array format.
+		// TODO add get_fields or similar method to get things out of the table.
+		// TODO extract this configuration in a better way, still make it work with `$wpdb` names.
+		// TODO take over the replicastore functions and move them over to this class.
+		// TODO make the API work.
+
+		$this->allowed_tables = apply_filters( 'jetpack_sync_checksum_allowed_tables', $this->default_tables );
+
+		$this->table               = $this->validate_table_name( $table );
+		$this->table_configuration = $this->allowed_tables[ $table ];
+
+		$this->prepare_fields( $this->table_configuration );
+
+	}
+
+	/**
+	 * Get Default Table configurations.
+	 *
+	 * @return array
+	 */
+	private function get_default_tables() {
+		global $wpdb;
+
+		 return array(
+			 'posts'              => array(
+				 'table'           => $wpdb->posts,
+				 'range_field'     => 'ID',
+				 'key_fields'      => array( 'ID' ),
+				 'checksum_fields' => array( 'post_modified_gmt' ),
+				 'filter_sql'      => Settings::get_blacklisted_post_types_sql(),
+			 ),
+			 'postmeta'           => array(
+				 'table'           => $wpdb->postmeta,
+				 'range_field'     => 'post_id',
+				 'key_fields'      => array( 'post_id', 'meta_key' ),
+				 'checksum_fields' => array( 'meta_key', 'meta_value' ),
+				 'filter_sql'      => Settings::get_whitelisted_post_meta_sql(),
+			 ),
+			 'comments'           => array(
+				 'table'           => $wpdb->comments,
+				 'range_field'     => 'comment_ID',
+				 'key_fields'      => array( 'comment_ID' ),
+				 'checksum_fields' => array( 'comment_content' ),
+				 'filter_sql'      => Settings::get_comments_filter_sql(),
```

Comment:
This filter is not enough, we also need to include
https://opengrok.a8c.com/source/xref/jetpack/packages/sync/src/modules/class-comments.php?r=5879967a#183
get_whitelisted_comment_types

Since we only send to WP.com allowed comment_types.

Reply:
Solved in cff37e6 , albeit not very prettily. We might want to extract allow/disallow lists/actions outside of the modules themselves or register for checksums only when the specific module is enabled. 

The latter seems like the better idea. We might want to do that for V2.